vllm/worker/model_runner.py

```diff
@@ -365,20 +365,45 @@ def _compute_for_prefix_cache_hit(
                             and self.sliding_window is None
                             and inter_data.is_prompt)
         inter_data.prefix_cache_hit = prefix_cache_hit
-        if self.chunked_prefill_enabled and prefix_cache_hit:
+        if (self.chunked_prefill_enabled and prefix_cache_hit
+                and self.attn_backend.get_name() != "flash-attn"):
             raise RuntimeError(
-                "chunked prefill cannot be used with prefix caching now.")
+                "Chunked prefill and prefix caching can only be used "
+                "simultaneously with flash-attn backend, try switching "
+                "to flash-attn backend by setting the environment variable "
+                "\"VLLM_ATTENTION_BACKEND=FLASH_ATTN\"")
```

Comment:
`prepare_input` is now backend agnostic. Can you put this checker to `vllm/attention/backends/flashinfer.py` and `vllm/attention/backends/utils.py`? For example in utils.py:

```python
    def _add_seq_group(
            self, inter_data: "ModelInputForGPUBuilder.InterDataForSeqGroup",
            chunked_prefill_enabled: bool):
        is_prompt = inter_data.is_prompt
        block_tables = inter_data.block_tables
        computed_block_nums = inter_data.computed_block_nums

        if chunked_prefill_enabled and inter_data.prefix_cache_hit:
            raise RuntimerError(...)
```